akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingGracefulShutdownSpec.scala

```diff
@@ -87,76 +35,41 @@ class PersistentClusterShardingGracefulShutdownMultiJvmNode2 extends PersistentC
-
-  val storageLocations = List(
-    new File(system.settings.config.getString("akka.cluster.sharding.distributed-data.durable.lmdb.dir")).getParentFile)
-
-  override protected def atStartup(): Unit = {
-    storageLocations.foreach(dir => if (dir.exists) FileUtils.deleteQuietly(dir))
-    enterBarrier("startup")
-  }
 
-  override protected def afterTermination(): Unit = {
-    storageLocations.foreach(dir => if (dir.exists) FileUtils.deleteQuietly(dir))
-  }
+  import MultiNodeClusterShardingSpec.ShardedEntity
+  import multiNodeConfig._
 
-  def join(from: RoleName, to: RoleName): Unit = {
+  def join(from: RoleName, to: RoleName, typeName: String): Unit = {
+    super.join(from, to)
     runOn(from) {
-      Cluster(system).join(node(to).address)
-      startSharding()
+      startSharding(typeName)
     }
-    enterBarrier(from.name + "-joined")
+    enterBarrier(s"$from-started")
   }
 
-  def startSharding(): Unit = {
-    val allocationStrategy =
-      new ShardCoordinator.LeastShardAllocationStrategy(rebalanceThreshold = 2, maxSimultaneousRebalance = 1)
-    ClusterSharding(system).start(
-      typeName = "Entity",
-      entityProps = Props[Entity],
-      settings = ClusterShardingSettings(system),
-      extractEntityId = extractEntityId,
-      extractShardId = extractShardId,
-      allocationStrategy,
-      handOffStopMessage = StopEntity)
-  }
+  def startSharding(typeName: String): ActorRef =
+    startSharding(
+      system,
+      typeName,
+      entityProps = Props[ShardedEntity],
+      extractEntityId = MultiNodeClusterShardingSpec.intExtractEntityId,
+      extractShardId = MultiNodeClusterShardingSpec.intExtractShardId,
+      allocationStrategy =
+        new ShardCoordinator.LeastShardAllocationStrategy(rebalanceThreshold = 2, maxSimultaneousRebalance = 1),
+      handOffStopMessage = ShardedEntity.Stop)
 
   lazy val region = ClusterSharding(system).shardRegion("Entity")
 
-  def isDdataMode: Boolean = mode == ClusterShardingSettings.StateStoreModeDData
-
   s"Cluster sharding ($mode)" must {
 
-    if (!isDdataMode) {
-      "setup shared journal" in {
-        // start the Persistence extension
-        Persistence(system)
-        runOn(first) {
-          system.actorOf(Props[SharedLeveldbStore], "store")
-        }
-        enterBarrier("peristence-started")
-
-        runOn(first, second) {
-          system.actorSelection(node(first) / "user" / "store") ! Identify(None)
-          val sharedStore = expectMsgType[ActorIdentity](10.seconds).ref.get
-          SharedLeveldbJournal.setStore(sharedStore, system)
-        }
-
-        enterBarrier("after-1")
-      }
-    }
-
     "start some shards in both regions" in within(30.seconds) {
-      join(first, first)
-      join(second, first)
+      startPersistenceIfNotDdataMode(startOn = first, setStoreOn = Seq(first, second))
+
+      join(first, first, typeName = "Entity")
+      join(second, first, typeName = "Entity")
```

Comment:
Perhaps extract this typeName to a field and reference that in the (3?) places it is now hardcoded.

Reply:
Done